src/main/java/com/liberty52/product/global/util/Result.java

```diff
@@ -148,16 +155,23 @@ public String toString() {
 
     public static final class Failure<T> extends Result<T> {
         private final Throwable error;
+        private final RuntimeException runtimeException;
 
         public Failure(Throwable throwable) {
             this.error = throwable;
+            this.runtimeException = (RuntimeException) throwable;
         }
```

Comment:
`Throwable`을 `RuntimeException`으로 다운캐스팅 할 때 캐스팅 에러가 발생할 수 있을 거 같아요.
그냥 `RuntimeException`만 필드로 가지고 생성자로 RuntimeException을 받아서 초기화하는 게 나아 보입니다.

```suggestion
    public static final class Failure<T> extends Result<T> {
        private final RuntimeException runtimeException;

        public Failure(RuntimException runtimeException ) {
            this.runtimeException = runtimeException;
        }
```
kotlin에선 try catch가 없어서 Thrawable로 해도 되는데 java에선 Throwable이면 무조건 캐치 해야 한다는 걸 깜빡했네요.

nit. `getOrThrow` 없애고 `getOrThrows`를 `getOrThrow`로 메서드 이름 바꾸면 될 듯 해요

Reply:
@jinlee0 
그럼 혹시 RuntimeException이 아닌 Exception의 경우에는 대응되지 못할 수도 있으니, 이런 방향은 어떤가요?

```java
        private final RuntimeException error;

        public Failure(Throwable throwable) {
            if (throwable instanceof RuntimeException e) {
                this.error = e;
            } else {
                this.error = new RuntimeException(throwable);
            }
        }
```

Reply:
LGTM